pkg/e2e/osd/machinehealthcheck.go

```diff
@@ -0,0 +1,88 @@
+package osd
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	machineV1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	OperatorNamespace = "openshift-machine-api"
+)
+
+var _ = ginkgo.Describe("[Suite: informing] MachineHealthChecks", func() {
+	h := helper.New()
+
+	ginkgo.It("should exist", func() {
+		machineHealthChecks, err := h.Machine().MachineV1beta1().MachineHealthChecks(OperatorNamespace).List(context.TODO(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, mhc := range machineHealthChecks.Items {
+			// verify there's an MHC for infra and worker nodes
+			Expect(mhc.Spec.Selector.MatchLabels).To(SatisfyAny(
+				Equal("machine.openshift.io/cluster-api-machine-role: infra"),
+				Equal("machine.openshift.io/cluster-api-machine-role: worker")),
+			)
+
+			// verify the unhealthy conditions are on all nodes
+			Expect(mhc.Spec.UnhealthyConditions).To(SatisfyAll(
+				ContainElement(
+					machineV1beta1.UnhealthyCondition{
+						Type:    corev1.NodeReady,
+						Status:  corev1.ConditionFalse,
+						Timeout: "480",
+					},
+				),
+				ContainElement(
+					machineV1beta1.UnhealthyCondition{
+						Type:    corev1.NodeReady,
+						Status:  corev1.ConditionUnknown,
+						Timeout: "480",
+					},
+				),
+			))
+		}
+	})
+
+	ginkgo.It("should replace unhealthy nodes", func() {
+		r := h.Runner("chroot /host -- systemctl stop kubelet")
+		r.Name = "stop-kubelet"
+		// i can't believe SecurityContext.Privileged is a pointer to a bool
+		thisIsSoDumb := true
```

Comment:
Nit: I can't believe it either, but in cases like this I typically name the variable `truePointer` ;)

Reply:
that's a much more descriptive variable name that still maintains the silliness, changed :)